VSNetBeans build fails on java.mx.project missing tag in build.xml

### Apache NetBeans version

Apache NetBeans 16

### What happened

VSNetBeans builder failed at https://ci-builds.apache.org/job/Netbeans/job/netbeans-vscode/1389/consoleText
at bottom there is following error:
```
-checkout-graalvm:
     [exec] Cloning into 'graal'...
     [exec] warning: Could not find remote branch vm-20.3.0 to clone.
     [exec] fatal: Remote branch vm-20.3.0 not found in upstream origin

BUILD FAILED
/home/jenkins/workspace/Netbeans/netbeans-vscode/java/java.mx.project/build.xml:30: exec returned: 128
```
Checking https://github.com/oracle/graal git repo there is no [`vm-20.3.0`](https://github.com/apache/netbeans/blob/6e18d439dbcf608714abe1f5b855900db9fd6216/java/java.mx.project/build.xml#L34) tag nor branch. Similarly there is also a reference to mx branch [`5.279.0`](https://github.com/apache/netbeans/blob/6e18d439dbcf608714abe1f5b855900db9fd6216/java/java.mx.project/build.xml#L44) at the same build.xml.
The failure happened with build #1389, VSIX itself is built.

### How to reproduce

See the build on jenkins https://ci-builds.apache.org/job/Netbeans/job/netbeans-vscode/1389/

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

ubuntu jenkins builder

### JDK

JDK11

### Apache NetBeans packaging

Apache VSNetBeans for VSCode

### Anything else

_No response_

### Are you willing to submit a pull request?

Yes

### Code of Conduct

Yes

--- a/java/java.mx.project/build.xml
+++ b/java/java.mx.project/build.xml
@@ -26,15 +26,31 @@
 
     <target name="-checkout-graalvm" unless="graal.dir.available">
         <delete dir="${graal.dir}"/>
-        <mkdir dir="${graal.dir}/.."/>
-        <exec dir="${graal.dir}/.." executable="git" failonerror="true">
-            <arg value="clone"/>
-            <arg value="--depth=1"/>
-            <arg value="--branch"/>
-            <arg value="vm-20.3.0"/>
-            <arg value="--single-branch"/>
+        <mkdir dir="${graal.dir}"/>
+
+        <!-- Avoid git-clone of the full Graal repository -->
+        <exec dir="${graal.dir}" executable="git" failonerror="true">
+            <arg value="init"/>
+        </exec>
+        <exec dir="${graal.dir}" executable="git" failonerror="true">
+            <arg value="remote"/>
+            <arg value="add"/>
+            <arg value="origin"/>
             <arg value="https://github.com/oracle/graal"/>
         </exec>
+        <echo message="Fetching GraalVM revision from https://github.com/oracle/graal"/>
+        <exec dir="${graal.dir}" executable="git" failonerror="true">
+            <arg value="fetch"/>
+            <arg value="origin"/>
+            <!-- The commit ID corresponds to a removed tag vm-20.3.0, by Gilles Duboscq, message "Release 20.3.0", Thu Nov 12 22:49:14 -->
+            <arg value="c5ff5a5a091bfd974640aaf6fbe51c81bd080438"/>
+        </exec>
+        <exec dir="${graal.dir}" executable="git" failonerror="true">
+            <arg value="reset"/>
+            <arg value="--hard"/>
+            <arg value="FETCH_HEAD"/>
+        </exec>
+
         <delete dir="${mx.dir}"/>
         <mkdir dir="${mx.dir}/.."/>
         <exec dir="${mx.dir}/.." executable="git" failonerror="true">

